exporter/stackdriverexporter/factory.go

```diff
@@ -0,0 +1,31 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package stackdriverexporter
+
+import (
+	"go.opentelemetry.io/collector/component"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/googlecloudexporter"
+)
+
+const (
+	// The value of "type" key in configuration.
+	typeStr = "stackdriver"
+)
+
+// NewFactory creates a factory for the stackdriver exporter
+func NewFactory() component.ExporterFactory {
+	return googlecloudexporter.NewFactoryWithTypeStr(typeStr)
+}
```

Comment:
instead of exposing `NewFactoryWithTypeStr` you can use composition and overload the funcs:

```golang
const (
	// The value of "type" key in configuration.
	typeStr = "stackdriver"
)
type factory struct {
  delegate component.ExporterFactory
}

// NewFactory creates a factory for the stackdriver exporter
func NewFactory() component.ExporterFactory {
	return &factory{ delegate: googlecloudexporter.NewFactoryWithTypeStr(typeStr)}
}

func (*factory) Type() configmodels.Type {
  return type
}

func (*factory) CreateDefaultConfig() cofigmodels.Exporter {
  cfg := delegate.CreateDefaultConfig()
  cfg.(googlecloud.Config).TypeVal = type
}
```

Reply:
@bogdandrutu I updated it, is that what you had in mind?